{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Dashboard notebook to control shared dask cluster "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/bokeh/core.py:57: UserWarning: \n",
      "Port 8787 is already in use. \n",
      "Perhaps you already have a cluster running?\n",
      "Hosting the diagnostics dashboard on a random port instead.\n",
      "  warnings.warn('\\n' + msg)\n"
     ]
    }
   ],
   "source": [
    "from dask_jobqueue import PBSCluster\n",
    "from distributed import Client\n",
    "\n",
    "cluster = PBSCluster(project='P48500028',\n",
    "                     queue='premium',\n",
    "                     walltime='02:00:00')\n",
    "\n",
    "client = Client(cluster)\n",
    "\n",
    "# cluster.scale(4)  # TODO: figure out why we need to scale up to 1 worker to open the bcsd/maurer datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "d0145eada9ee486389726d418d233b75",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "VBox(children=(HTML(value='<h2>PBSCluster</h2>'), HBox(children=(HTML(value='\\n<div>\\n  <style scoped>\\n    .dâ€¦"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<distributed.deploy.adaptive.Adaptive at 0x2aab659080b8>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster.adapt(minimum=72, maximum=90, interval='500ms')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table style=\"border: 2px solid white;\">\n",
       "<tr>\n",
       "<td style=\"vertical-align: top; border: 0px solid white\">\n",
       "<h3>Client</h3>\n",
       "<ul>\n",
       "  <li><b>Scheduler: </b>tcp://10.148.10.31:47938\n",
       "  <li><b>Dashboard: </b><a href='proxy/36866/status' target='_blank'>proxy/36866/status</a>\n",
       "</ul>\n",
       "</td>\n",
       "<td style=\"vertical-align: top; border: 0px solid white\">\n",
       "<h3>Cluster</h3>\n",
       "<ul>\n",
       "  <li><b>Workers: </b>72</li>\n",
       "  <li><b>Cores: </b>288</li>\n",
       "  <li><b>Memory: </b>360.00 GB</li>\n",
       "</ul>\n",
       "</td>\n",
       "</tr>\n",
       "</table>"
      ],
      "text/plain": [
       "<Client: scheduler='tcp://10.148.10.31:47938' processes=72 cores=288>"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Worker tcp://10.148.13.152:50152 restart in Job 4773132. This can be due to memory issue.\n",
      "Worker tcp://10.148.3.76:39976 restart in Job 4773145. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.77:36962 restart in Job 4773129. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.75:54503 restart in Job 4773127. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.126:33354 restart in Job 4773136. This can be due to memory issue.\n",
      "Worker tcp://10.148.8.175:35269 restart in Job 4773126. This can be due to memory issue.\n",
      "Worker tcp://10.148.8.24:46348 restart in Job 4773125. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.183:54921 restart in Job 4773266. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.183:39953 restart in Job 4773266. This can be due to memory issue.\n",
      "Worker tcp://10.148.8.24:49421 restart in Job 4773125. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.75:50502 restart in Job 4773127. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.152:56753 restart in Job 4773132. This can be due to memory issue.\n",
      "Worker tcp://10.148.3.76:42201 restart in Job 4773145. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.77:38421 restart in Job 4773129. This can be due to memory issue.\n",
      "Worker tcp://10.148.8.175:50416 restart in Job 4773126. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.126:40089 restart in Job 4773136. This can be due to memory issue.\n",
      "Worker tcp://10.148.14.117:46513 restart in Job 4774070. This can be due to memory issue.\n",
      "distributed.utils - ERROR - 'tcp://10.148.14.117:46513'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/utils.py\", line 649, in log_errors\n",
      "    yield\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:46513'\n",
      "distributed.utils - ERROR - 'tcp://10.148.14.117:54655'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/utils.py\", line 649, in log_errors\n",
      "    yield\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:54655'\n",
      "distributed.utils - ERROR - 'tcp://10.148.14.117:43360'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/utils.py\", line 649, in log_errors\n",
      "    yield\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:43360'\n",
      "distributed.utils - ERROR - 'tcp://10.148.14.117:35930'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/utils.py\", line 649, in log_errors\n",
      "    yield\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:35930'\n",
      "distributed.utils - ERROR - 'tcp://10.148.14.117:41841'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/utils.py\", line 649, in log_errors\n",
      "    yield\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:41841'\n",
      "distributed.utils - ERROR - 'tcp://10.148.14.117:52855'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/utils.py\", line 649, in log_errors\n",
      "    yield\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:52855'\n",
      "distributed.utils - ERROR - 'tcp://10.148.14.117:53160'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/utils.py\", line 649, in log_errors\n",
      "    yield\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:53160'\n",
      "distributed.utils - ERROR - 'tcp://10.148.14.117:42273'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/utils.py\", line 649, in log_errors\n",
      "    yield\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:42273'\n",
      "distributed.utils - ERROR - 'tcp://10.148.14.117:35743'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/utils.py\", line 649, in log_errors\n",
      "    yield\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:35743'\n",
      "distributed.core - ERROR - 'tcp://10.148.14.117:46513'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/core.py\", line 346, in handle_comm\n",
      "    result = yield result\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1141, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:46513'\n",
      "distributed.core - ERROR - 'tcp://10.148.14.117:54655'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/core.py\", line 346, in handle_comm\n",
      "    result = yield result\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1141, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:54655'\n",
      "distributed.core - ERROR - 'tcp://10.148.14.117:43360'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/core.py\", line 346, in handle_comm\n",
      "    result = yield result\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1141, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:43360'\n",
      "distributed.core - ERROR - 'tcp://10.148.14.117:35930'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/core.py\", line 346, in handle_comm\n",
      "    result = yield result\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1141, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:35930'\n",
      "distributed.core - ERROR - 'tcp://10.148.14.117:41841'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/core.py\", line 346, in handle_comm\n",
      "    result = yield result\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1141, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:41841'\n",
      "distributed.core - ERROR - 'tcp://10.148.14.117:52855'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/core.py\", line 346, in handle_comm\n",
      "    result = yield result\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1141, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:52855'\n",
      "distributed.core - ERROR - 'tcp://10.148.14.117:53160'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/core.py\", line 346, in handle_comm\n",
      "    result = yield result\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1141, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:53160'\n",
      "distributed.core - ERROR - 'tcp://10.148.14.117:42273'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/core.py\", line 346, in handle_comm\n",
      "    result = yield result\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1141, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:42273'\n",
      "distributed.core - ERROR - 'tcp://10.148.14.117:35743'\n",
      "Traceback (most recent call last):\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/core.py\", line 346, in handle_comm\n",
      "    result = yield result\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1141, in run\n",
      "    yielded = self.gen.throw(*exc_info)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 1359, in add_worker\n",
      "    yield self.handle_worker(comm=comm, worker=address)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 1133, in run\n",
      "    value = future.result()\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/tornado/gen.py\", line 326, in wrapper\n",
      "    yielded = next(result)\n",
      "  File \"/glade/u/home/jvano/miniconda/envs/loca/lib/python3.6/site-packages/distributed/scheduler.py\", line 2219, in handle_worker\n",
      "    worker_comm = self.stream_comms[worker]\n",
      "KeyError: 'tcp://10.148.14.117:35743'\n",
      "Worker tcp://10.148.13.75:50502 restart in Job 4773127. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.177:50788 restart in Job 4774043. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.152:56753 restart in Job 4773132. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.77:38421 restart in Job 4773129. This can be due to memory issue.\n",
      "Worker tcp://10.148.3.76:42201 restart in Job 4773145. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.183:36668 restart in Job 4773266. This can be due to memory issue.\n",
      "Worker tcp://10.148.12.136:57761 restart in Job 4774058. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.126:52525 restart in Job 4773136. This can be due to memory issue.\n",
      "Worker tcp://10.148.12.136:42001 restart in Job 4774058. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.177:59245 restart in Job 4774043. This can be due to memory issue.\n",
      "Worker tcp://10.148.3.76:41840 restart in Job 4773145. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.75:60639 restart in Job 4773127. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.126:55332 restart in Job 4773136. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.183:38692 restart in Job 4773266. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.77:58044 restart in Job 4773129. This can be due to memory issue.\n",
      "Worker tcp://10.148.13.152:47046 restart in Job 4773132. This can be due to memory issue.\n",
      "distributed.utils_perf - WARNING - full garbage collections took 13% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 16% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 16% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 16% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 16% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 16% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 17% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 17% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 17% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 17% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 17% CPU time recently (threshold: 10%)\n",
      "distributed.utils_perf - WARNING - full garbage collections took 16% CPU time recently (threshold: 10%)\n"
     ]
    }
   ],
   "source": [
    "client"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import os\n",
    "with open(os.path.join(os.environ['HOME'], 'scheduler_file.json'), 'w') as f:\n",
    "    json.dump(client.scheduler_info(), f, indent=2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "INSTALLED VERSIONS\n",
      "------------------\n",
      "commit: None\n",
      "python: 3.6.7 | packaged by conda-forge | (default, Feb 28 2019, 09:07:38) \n",
      "[GCC 7.3.0]\n",
      "python-bits: 64\n",
      "OS: Linux\n",
      "OS-release: 3.12.62-60.64.8-default\n",
      "machine: x86_64\n",
      "processor: x86_64\n",
      "byteorder: little\n",
      "LC_ALL: None\n",
      "LANG: en_US.UTF-8\n",
      "LOCALE: en_US.UTF-8\n",
      "libhdf5: 1.10.4\n",
      "libnetcdf: 4.6.2\n",
      "\n",
      "xarray: 0.11.3\n",
      "pandas: 0.24.1\n",
      "numpy: 1.16.2\n",
      "scipy: 1.2.1\n",
      "netCDF4: 1.4.2\n",
      "pydap: None\n",
      "h5netcdf: 0.7.0\n",
      "h5py: 2.9.0\n",
      "Nio: None\n",
      "zarr: None\n",
      "cftime: 1.0.3.4\n",
      "PseudonetCDF: None\n",
      "rasterio: None\n",
      "cfgrib: None\n",
      "iris: None\n",
      "bottleneck: 1.2.1\n",
      "cyordereddict: None\n",
      "dask: 1.1.2\n",
      "distributed: 1.26.0\n",
      "matplotlib: 2.2.3\n",
      "cartopy: 0.17.0\n",
      "seaborn: 0.9.0\n",
      "setuptools: 40.8.0\n",
      "pip: 19.0.3\n",
      "conda: None\n",
      "pytest: None\n",
      "IPython: 7.3.0\n",
      "sphinx: None\n"
     ]
    }
   ],
   "source": [
    "import xarray as xr\n",
    "xr.show_versions()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:loca]",
   "language": "python",
   "name": "conda-env-loca-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
